src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -833,10 +833,11 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
 
     ucs_debug("%s: opened DEVX md", ibv_get_device_name(ibv_device));
 
-    dev->flags |= UCT_IB_DEVICE_FLAG_MLX5_PRM;
-    md->flags  |= UCT_IB_MLX5_MD_FLAG_DEVX;
-    md->flags  |= UCT_IB_MLX5_MD_FLAGS_DEVX_OBJS(md_config->devx_objs);
-    *p_md       = &md->super;
+    dev->flags      |= UCT_IB_DEVICE_FLAG_MLX5_PRM;
+    md->flags       |= UCT_IB_MLX5_MD_FLAG_DEVX;
+    md->flags       |= UCT_IB_MLX5_MD_FLAGS_DEVX_OBJS(md_config->devx_objs);
+    md->super.flags |= UCT_IB_MD_FLAG_MLX5;
```

Comment:
how is it different from dev->flags?

Reply:
so, can we check dev->flags instead?

Reply:
seems like yes, since UCT_IB_DEVICE_FLAG_MLX5_PRM is set explicitly when mlx5 md is opened, but for some reason it didn't work as part of query tl resources - need to check why

Reply:
didn't work due to https://github.com/openucx/ucx/blob/7d8e496e9b397bfb7a5a4cd79f2497cf8bdd94f6/src/uct/ib/base/ib_md.c#L1745
flag UCT_IB_DEVICE_FLAG_MLX5_PRM is set there

Reply:
seems that need to make the md->dev.flags initialization consistent across mlx5/verbs MDs

Reply:
do you mean remove UCT_IB_DEVICE_FLAG_MLX5_PRM flags from uct_ib_builtin_device_specs?

Reply:
or clear this flag in uct_ib_verbs_md_open function?